determined-ai/determined
# feat: edit and reset raw user settings [WEB-1361]

## Description
Note: final implementation of this is blocked by the [settings drawer PR](https://github.com/determined-ai/determined/pull/7356), for now the buttons appear on the dashboard to test functionality.




## Test Plan
Since this PR involves backend changes, the submitting of edited settings won't work on netlify. Browsing and resetting should work as expected.

    1. Delete setting in modal editor
      1a. Confirm that setting remains deleted in editor after submission
      1b. Confirm setting becomes default in the app
      1c. Confirm it persists on reload
    2. Change a setting in the editor
      2a. Confirm that setting remains changed in editor after submission
      2b. Confirm setting changes in the app
      2c. Confirm it persists on reload
    3. Change a setting to something invalid for that setting (but still JSON)
      3a. Confirm submission works
      3b. Confirm that it is treated as if the setting is not set in the app
    4. Hit Reset Settings
      4a. Confirm modal is updated
      4b. Confirm app is updated
      4c. Confirm reset persists after refresh
    5. Write invalid JSON in the editor
      5a. Confirm you receive a validation error and the form is not submitted.






## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
